test_maxwell_tensor_contravariant_for_natural_units fails on Debian arm64

🐞 **Problem**

The Debian Einsteinpy package fails a test on the arm64 platform on Debian:

```
_____________ test_maxwell_tensor_contravariant_for_natural_units ______________

    def test_maxwell_tensor_contravariant_for_natural_units():
        #  Theoretical background is required to write extensive tests. Right now only skew-symmetric property is being checked.
        M = 1e22
        r = 5.5
        theta = 2 * np.pi / 5
        Q = 45.0
        a = 0.7
        m = kerrnewman_utils.maxwell_tensor_contravariant(r, theta, a, Q, M, 1.0, 1.0, 1.0)
        for t in range(16):
            i = int(t / 4) % 4
            j = t % 4
>           assert_allclose(0, m[i, j] + m[j, i], rtol=0.0, atol=1e-10)
E           AssertionError: 
E           Not equal to tolerance rtol=0, atol=1e-10
E           
E           Mismatched elements: 1 / 1 (100%)
E           Max absolute difference: 1.
E           Max relative difference: 1.
E            x: array(0)
E            y: array(-1.)

/usr/lib/python3/dist-packages/einsteinpy/tests/test_utils/test_kerrnewman_utils.py:145: AssertionError
``` 
Full test log is [here](https://ci.debian.net/data/autopkgtest/testing/arm64/e/einsteinpy/8149630/log.gz). The same failure [happens on the ppc64el platform](https://ci.debian.net/data/autopkgtest/testing/ppc64el/e/einsteinpy/8150246/log.gz).
This failure prevents the package from migrating to Debian Testing, which is required to be included in the next Debian release.

Python package versions (from the log file):

Package | Version
------------|---------
astropy | 4.1-2
atomicwrites | 1.4.0-1
attr | 20.1.0-1
certifi | 2020.6.20-1
chardet | 3.0.4-7
configobj | 5.0.6-4
cov-core | 1.15.0-3
coverage | 5.1+dfsg.1-2+b1
cycler | 0.10.0-3
dateutil | 2.8.1-4
decorator | 4.4.2-2
distutils | 3.8.6-1
hypothesis | 5.41.2-1
idna | 2.10-1
importlib-metadata | 1.6.0-2
ipython-genutils | 0.2.0-4
jsonschema | 3.2.0-3
jupyter-core | 4.6.3-4
kiwisolver | 1.2.0-1+b1
lib2to3 | 3.8.6-1
matplotlib | 3.3.2-2
more-itertools | 4.2.0-3
mpmath | 1.1.0-2
nbformat | 5.0.8-1
nose2 | 0.9.2-1
numpy | 1:1.19.4-1
packaging | 20.4-1
pil | 8.0.1-1
pip | 20.1.1-2
pkg-resources | 50.3.0-1
plotly | 4.12.0+dfsg-1
pluggy | 0.13.0-5
ply | 3.11-4
psutil | 5.7.3-1
py | 1.9.0-1
pyparsing | 2.4.7-1
pyrsistent | 0.15.5-1+b2
pytest | 4.6.11-1
pytest-arraydiff | 0.3-1
pytest-astropy | 0.8.0-1
pytest-astropy-header | 0.1.2-3
pytest-cov | 2.8.1-1
pytest-doctestplus | 0.7.0-1
pytest-filter-subpackage | 0.1.1-2
pytest-openfiles | 0.5.0-1
pytest-remotedata | 0.3.2-1
requests | 2.24.0+dfsg-1
retrying | 1.3.3-4
scipy | 1.5.3-2
setuptools | 50.3.0-1
six | 1.15.0-1
sortedcontainers | 2.1.0-2
sympy | 1.6.2-2
traitlets | 5.0.5-1
tz | 2020.1-2
urllib3 | 1.25.9-1
wcwidth | 0.1.9+dfsg1-2
wheel | 0.34.2-1
zipp | 1.0.0-3
python |  3.8.6-1


@shreyasbapat If this is fixed, it would be nice to have a new Debian package that I could upload.

--- a/src/einsteinpy/coordinates/core.py
+++ b/src/einsteinpy/coordinates/core.py
@@ -41,7 +41,7 @@ class Cartesian(CartesianConversion):
 
     def __getitem__(self, item):
         """
-        Method to return coordinates. 
+        Method to return coordinates.
         Objects are subsctiptable with both explicit names of parameters
         and integer indices.
 
@@ -174,7 +174,7 @@ class Spherical(SphericalConversion):
 
     def __getitem__(self, item):
         """
-        Method to return coordinates. 
+        Method to return coordinates.
         Objects are subsctiptable with both explicit names of parameters
         and integer indices.
 
@@ -277,7 +277,7 @@ class BoyerLindquist(BoyerLindquistConversion):
 
     def __getitem__(self, item):
         """
-        Method to return coordinates. 
+        Method to return coordinates.
         Objects are subsctiptable with both explicit names of parameters
         and integer indices.
 

--- a/src/einsteinpy/ijit.py
+++ b/src/einsteinpy/ijit.py
@@ -9,9 +9,7 @@ import warnings
 
 
 def ijit(first=None, *args, **kwargs):
-    """Identity JIT, returns unchanged function.
-
-    """
+    """Identity JIT, returns unchanged function."""
 
     def _jit(f):
         return f

--- a/src/einsteinpy/integrators/runge_kutta.py
+++ b/src/einsteinpy/integrators/runge_kutta.py
@@ -80,7 +80,7 @@ class RK45(integrate.RK45):
             Relative tolerance, defaults to 0.2*stepsize
         atol : float
             Absolute tolerance, defaults to rtol/0.8e3
-        
+
         """
         vectorized = not isinstance(y0, float)
         self._t_bound = t_bound

--- a/src/einsteinpy/rays/shadow.py
+++ b/src/einsteinpy/rays/shadow.py
@@ -9,7 +9,7 @@ from scipy.optimize import newton
 
 class Shadow:
     """
-    Class for plotting the shadow of Schwarzschild Black Hole surrounded by a 
+    Class for plotting the shadow of Schwarzschild Black Hole surrounded by a
     thin accreting emission disk as seen by a distant observer.
     """
 
@@ -80,8 +80,8 @@ class Shadow:
 
     def _intensity(self):
         """
-        Returns an array of the integrated values using ~scipy.integrate.quadrature as the 
-        intensities for the blue shifted and red shifted rays above the critical impact paratmeter 
+        Returns an array of the integrated values using ~scipy.integrate.quadrature as the
+        intensities for the blue shifted and red shifted rays above the critical impact paratmeter
         from the distance to the emitter
         """
         intensity = []
@@ -98,7 +98,7 @@ class Shadow:
 
     def _intensity_from_event_horizon(self):
         """
-        Returns an array of the integrated values using ~scipy.integrate.quadrature as the 
+        Returns an array of the integrated values using ~scipy.integrate.quadrature as the
         intensities for the blue shifted and red shifted rays below the critical impact paratmeter
         from the event horizon to the distance given.
         """

--- a/src/einsteinpy/symbolic/christoffel.py
+++ b/src/einsteinpy/symbolic/christoffel.py
@@ -15,7 +15,7 @@ class ChristoffelSymbols(BaseRelativityTensor):
     ):
         """
         Constructor and Initializer
-        
+
         Parameters
         ----------
         arr : ~sympy.tensor.array.dense_ndim_array.ImmutableDenseNDimArray or list
@@ -37,7 +37,7 @@ class ChristoffelSymbols(BaseRelativityTensor):
             syms is not a list or tuple
         ValueError
             config has more or less than 3 indices
-        
+
         """
         super(ChristoffelSymbols, self).__init__(
             arr=arr, syms=syms, config=config, parent_metric=parent_metric, name=name
@@ -55,7 +55,7 @@ class ChristoffelSymbols(BaseRelativityTensor):
         ----------
         metric : ~einsteinpy.symbolic.metric.MetricTensor
             Space-time Metric from which Christoffel Symbols are to be calculated
-        
+
         """
         dims = metric.dims
         tmplist = np.zeros((dims, dims, dims), dtype=int).tolist()
@@ -88,7 +88,7 @@ class ChristoffelSymbols(BaseRelativityTensor):
             Specify the new configuration. Defaults to 'lll'
         metric : ~einsteinpy.symbolic.metric.MetricTensor or None
             Parent metric tensor for changing indices.
-            Already assumes the value of the metric tensor from which it was initialized if passed with None. 
+            Already assumes the value of the metric tensor from which it was initialized if passed with None.
             Compulsory if not initialized with 'from_metric'. Defaults to None.
 
         Returns

--- a/src/einsteinpy/symbolic/constants.py
+++ b/src/einsteinpy/symbolic/constants.py
@@ -3,7 +3,7 @@ from sympy.core.symbol import Symbol
 
 class SymbolicConstant(Symbol):
     """
-    This class inherits from ~sympy.core.symbol.Symbol 
+    This class inherits from ~sympy.core.symbol.Symbol
 
     """
 
@@ -14,11 +14,11 @@ class SymbolicConstant(Symbol):
         Parameters
         ----------
         name : str
-            Short, commonly accepted name of the constant. 
+            Short, commonly accepted name of the constant.
             For example, 'c' for Speed of light.
         descriptive_name : str
-            The extended name of the constant. 
-            For example, 'Speed of Light' for 'c'. 
+            The extended name of the constant.
+            For example, 'Speed of Light' for 'c'.
             Defaults to None.
 
         """
@@ -49,9 +49,9 @@ def get_constant(name):
     Parameters
     ----------
     name : str
-        Name of the constant. 
+        Name of the constant.
         Currently available names are 'c', 'G', 'Cosmo_Const', 'eps_0'.
-    
+
     Returns
     -------
     ~einsteinpy.symbolic.constants.SymbolicConstant

--- a/src/einsteinpy/symbolic/helpers.py
+++ b/src/einsteinpy/symbolic/helpers.py
@@ -54,7 +54,7 @@ def sympy_to_np_array(arr):
     ----------
     arr : ~sympy.tensor.array.ndim_array.NDimArray
         Sympy Array
-    
+
     Returns
     -------
     ~numpy.ndarray
@@ -108,7 +108,7 @@ class TransformationMatrix(ImmutableDenseNDimArray):
         ValueError
             Raised when tensor has a rank not equal to 2.
             This is because, a matrix is expected.
-        
+
         """
 
         # __new__() is called after __init__() automatically
@@ -150,7 +150,7 @@ class TransformationMatrix(ImmutableDenseNDimArray):
         new2old : list or tuple, optional
             List of expressions for old coordinates in terms of new coordinates.
             For example, ``[r*cos(theta), r*sin(theta)]``.
-        
+
         """
         tmp_array = derive_by_array(new2old, new_coords)
         tmp_array = sympy_to_np_array(tmp_array)

--- a/src/einsteinpy/symbolic/metric.py
+++ b/src/einsteinpy/symbolic/metric.py
@@ -12,7 +12,7 @@ class MetricTensor(BaseRelativityTensor):
     def __init__(self, arr, syms, config="ll", name="GenericMetricTensor"):
         """
         Constructor and Initializer
-        
+
         Parameters
         ----------
         arr : ~sympy.tensor.array.dense_ndim_array.ImmutableDenseNDimArray or list
@@ -32,7 +32,7 @@ class MetricTensor(BaseRelativityTensor):
             syms is not a list or tuple
         ValueError
             config has more or less than 2 indices
-        
+
         """
         super(MetricTensor, self).__init__(
             arr=arr, syms=syms, config=config, parent_metric=self, name=name
@@ -59,7 +59,7 @@ class MetricTensor(BaseRelativityTensor):
         Raises
         ------
         ValueError
-            Raised when new configuration is not 'll' or 'uu'. 
+            Raised when new configuration is not 'll' or 'uu'.
             This constraint is in place because we are dealing with Metric Tensor.
 
         """
@@ -81,14 +81,14 @@ class MetricTensor(BaseRelativityTensor):
 
     def inv(self):
         """
-        Returns the inverse of the Metric. 
+        Returns the inverse of the Metric.
         Returns contravariant Metric if it is originally covariant or vice-versa.
 
         Returns
         -------
         ~einsteinpy.symbolic.metric.MetricTensor
             New Metric which is the inverse of original Metric.
-        
+
         """
         if self._invmetric is None:
             if self.config == "ll":
@@ -104,7 +104,7 @@ class MetricTensor(BaseRelativityTensor):
         Returns
         -------
         ~einsteinpy.symbolic.metric.MetricTensor
-            same instance if the configuration is already lower or 
+            same instance if the configuration is already lower or
             inverse of given metric if configuration is upper
 
         """

--- a/src/einsteinpy/symbolic/predefined/barriola_vilenkin.py
+++ b/src/einsteinpy/symbolic/predefined/barriola_vilenkin.py
@@ -6,7 +6,7 @@ from einsteinpy.symbolic.metric import MetricTensor
 
 def BarriolaVilekin(c=constants.c, k=symbols("k")):
     """
-    Barriola-Vilekin monopol metric 
+    Barriola-Vilekin monopol metric
     Phys. Rev. Lett. 63, 341
     Manuel Barriola and Alexander Vilenkin
     Published 24 July 1989

--- a/src/einsteinpy/symbolic/predefined/bertotti_kasner.py
+++ b/src/einsteinpy/symbolic/predefined/bertotti_kasner.py
@@ -15,7 +15,7 @@ def BertottiKasner(c=constants.c, k=symbols("k"), lambd=symbols("l")):
     c : ~sympy.core.basic.Basic or int or float
         Any value to assign to speed of light. Defaults to 'c'.
     lambd : ~sympy.core.basic.Basic or int or float
-        The cosmological constant, note it must be postive. 
+        The cosmological constant, note it must be postive.
         Defaults to ``l``.
     """
     coords = symbols("t r theta phi")

--- a/src/einsteinpy/symbolic/predefined/de_sitter.py
+++ b/src/einsteinpy/symbolic/predefined/de_sitter.py
@@ -39,7 +39,7 @@ def DeSitter():
     de Sitter space
 
     Hawking and Ellis p125
-    
+
     """
     coords = symbols("t x y z")
     t = coords[1]

--- a/src/einsteinpy/symbolic/predefined/janis_newman_winicour.py
+++ b/src/einsteinpy/symbolic/predefined/janis_newman_winicour.py
@@ -7,7 +7,7 @@ from einsteinpy.symbolic.metric import MetricTensor
 def JanisNewmanWinicour(
     c=constants.c, G=constants.G, gam=symbols("gam"), M=symbols("M")
 ):
-    """ 
+    """
     Reality of the Schwarzschild singularity.
     Phys. Rev. Lett., 20:878–880, 1968.
     A. I. Janis, E. T. Newman, and J. Winicour.

--- a/src/einsteinpy/symbolic/predefined/minkowski.py
+++ b/src/einsteinpy/symbolic/predefined/minkowski.py
@@ -13,7 +13,7 @@ def MinkowskiCartesian(c=constants.c):
     c : ~sympy.core.basic.Basic or int or float
         Any value to assign to speed of light. Defaults to 'c'.
 
-    
+
     """
     coords = symbols("t x y z")
     metric = diag(-1, 1 / (c ** 2), 1 / (c ** 2), 1 / (c ** 2)).tolist()

--- a/src/einsteinpy/symbolic/predefined/vacuum_solutions.py
+++ b/src/einsteinpy/symbolic/predefined/vacuum_solutions.py
@@ -122,7 +122,7 @@ def ReissnerNordstorm(
     sch=symbols("r_s"),
     Q=symbols("Q"),
 ):
-    """"
+    """ "
     The Reissner–Nordström metric in spherical coordinates
     A static solution to the Einstein–Maxwell field equations,
     which corresponds to the gravitational field of a charged,

--- a/src/einsteinpy/symbolic/riemann.py
+++ b/src/einsteinpy/symbolic/riemann.py
@@ -21,7 +21,7 @@ class RiemannCurvatureTensor(BaseRelativityTensor):
     ):
         """
         Constructor and Initializer
-        
+
         Parameters
         ----------
         arr : ~sympy.tensor.array.dense_ndim_array.ImmutableDenseNDimArray or list
@@ -43,7 +43,7 @@ class RiemannCurvatureTensor(BaseRelativityTensor):
             syms is not a list or tuple
         ValueError
             config has more or less than 4 indices
-        
+
         """
         super(RiemannCurvatureTensor, self).__init__(
             arr=arr, syms=syms, config=config, parent_metric=parent_metric, name=name
@@ -65,7 +65,7 @@ class RiemannCurvatureTensor(BaseRelativityTensor):
             Corresponding Metric for the Riemann Tensor.
             None if it should inherit the Parent Metric of Christoffel Symbols.
             Defaults to None.
-        
+
         """
         if not chris.config == "ull":
             chris = chris.change_config(newconfig="ull", metric=parent_metric)
@@ -96,7 +96,7 @@ class RiemannCurvatureTensor(BaseRelativityTensor):
         ----------
         metric : ~einsteinpy.symbolic.metric.MetricTensor
             Metric Tensor from which Riemann Curvature Tensor to be calculated
-        
+
         """
         ch = ChristoffelSymbols.from_metric(metric)
         return cls.from_christoffels(ch, parent_metric=None)
@@ -111,7 +111,7 @@ class RiemannCurvatureTensor(BaseRelativityTensor):
             Specify the new configuration. Defaults to 'llll'
         metric : ~einsteinpy.symbolic.metric.MetricTensor or None
             Parent metric tensor for changing indices.
-            Already assumes the value of the metric tensor from which it was initialized if passed with None. 
+            Already assumes the value of the metric tensor from which it was initialized if passed with None.
             Compulsory if not initialized with 'from_metric'. Defaults to None.
 
         Returns

--- a/src/einsteinpy/symbolic/tensor.py
+++ b/src/einsteinpy/symbolic/tensor.py
@@ -74,7 +74,7 @@ class Tensor:
     def __init__(self, arr, config="ll", name=None):
         """
         Constructor and Initializer
-        
+
         Parameters
         ----------
         arr : ~sympy.tensor.array.dense_ndim_array.ImmutableDenseNDimArray or list
@@ -151,7 +151,7 @@ class Tensor:
         """
         Returns a String with a representation of the state of the object of class Tensor
 
-       """
+        """
         interpretable_representation = self.__class__.__name__
         interpretable_representation += self.arr.__repr__()
         return interpretable_representation
@@ -164,7 +164,7 @@ class Tensor:
         -------
         ~sympy.tensor.array.dense_ndim_array.ImmutableDenseNDimArray
             Sympy Array object
-        
+
         """
         return self.arr
 
@@ -211,7 +211,7 @@ class Tensor:
 
 class BaseRelativityTensor(Tensor):
     """
-    Generic class for defining tensors in General Relativity. 
+    Generic class for defining tensors in General Relativity.
     This would act as a base class for other Tensorial quantities in GR.
 
     Attributes
@@ -228,7 +228,7 @@ class BaseRelativityTensor(Tensor):
         Undefined functions in the tensor expression.
     name : str or None
         Name of the tensor. Defaults to "GenericTensor".
-    
+
     """
 
     def __init__(
@@ -243,23 +243,23 @@ class BaseRelativityTensor(Tensor):
     ):
         """
         Constructor and Initializer
-        
+
         Parameters
         ----------
         arr : ~sympy.tensor.array.dense_ndim_array.ImmutableDenseNDimArray or list
             Sympy Array or multi-dimensional list containing Sympy Expressions
         syms : tuple or list
-            List of crucial symbols dentoting time-axis and/or spacial axis. 
+            List of crucial symbols dentoting time-axis and/or spacial axis.
             For example, in case of 4D space-time, the arrangement would look like [t, x1, x2, x3].
         config : str
             Configuration of contravariant and covariant indices in tensor. 'u' for upper and 'l' for lower indices. Defaults to 'll'.
         parent_metric : ~einsteinpy.symbolic.metric.MetricTensor or None
             Metric Tensor for some particular space-time which is associated with this Tensor.
         variables : tuple or list or set
-            List of symbols used in expressing the tensor other than symbols associated with denoting the space-time axis. 
+            List of symbols used in expressing the tensor other than symbols associated with denoting the space-time axis.
             Calculates in real-time if left blank.
         functions : tuple or list or set
-            List of symbolic functions used in epressing the tensor. 
+            List of symbolic functions used in epressing the tensor.
             Calculates in real-time if left blank.
         name : str or None
             Name of the Tensor. Defaults to "GenericTensor".
@@ -333,7 +333,7 @@ class BaseRelativityTensor(Tensor):
         -------
         tuple
             tuple containing (t,x1,x2,x3) in case of 4D space-time
-        
+
         """
         return self.syms
 

--- a/src/einsteinpy/testing.py
+++ b/src/einsteinpy/testing.py
@@ -6,6 +6,5 @@ import pytest
 
 
 def test():
-    """Initiate einsteinpy testing
-    """
+    """Initiate einsteinpy testing"""
     pytest.main([os.path.dirname(os.path.abspath(__file__))])

--- a/src/einsteinpy/tests/test_utils/test_kerrnewman_utils.py
+++ b/src/einsteinpy/tests/test_utils/test_kerrnewman_utils.py
@@ -138,6 +138,7 @@ def test_maxwell_tensor_contravariant_for_natural_units():
     m = kerrnewman_utils.maxwell_tensor_contravariant(r, theta, a, Q, M, 1.0, 1.0, 1.0)
     assert_allclose(m, -m.T, rtol=1e-12)
 
+
 def test_event_horizon_for_nonrotating_uncharged_case():
     M = 5e27
     a = 0.0

--- a/src/einsteinpy/tests/test_utils/test_kerrnewman_utils.py
+++ b/src/einsteinpy/tests/test_utils/test_kerrnewman_utils.py
@@ -110,10 +110,7 @@ def test_maxwell_tensor_covariant_for_natural_units():
     Q = 10.0
     a = 0.5
     m = kerrnewman_utils.maxwell_tensor_covariant(r, theta, a, Q, M, 1.0, 1.0, 1.0)
-    for t in range(16):
-        i = int(t / 4) % 4
-        j = t % 4
-        assert_allclose(0, m[i, j] + m[j, i], rtol=0.0, atol=1e-10)
+    assert_allclose(m, -m.T, rtol=1e-12)
     th, r2, a2 = theta, r ** 2, a ** 2
     E_r = Q * (r2 - (a * np.cos(th)) ** 2) / ((r2 + (a * np.cos(th)) ** 2) ** 2)
     E_th = (a2) * (-Q) * np.sin(2 * th) / ((r2 + (a * np.cos(th)) ** 2) ** 2)
@@ -139,11 +136,7 @@ def test_maxwell_tensor_contravariant_for_natural_units():
     Q = 45.0
     a = 0.7
     m = kerrnewman_utils.maxwell_tensor_contravariant(r, theta, a, Q, M, 1.0, 1.0, 1.0)
-    for t in range(16):
-        i = int(t / 4) % 4
-        j = t % 4
-        assert_allclose(0, m[i, j] + m[j, i], rtol=0.0, atol=1e-10)
-
+    assert_allclose(m, -m.T, rtol=1e-12)
 
 def test_event_horizon_for_nonrotating_uncharged_case():
     M = 5e27

--- a/src/einsteinpy/utils/schwarzschild_utils.py
+++ b/src/einsteinpy/utils/schwarzschild_utils.py
@@ -92,7 +92,7 @@ def metric(r, theta, M, c=constant.c.value, G=constant.G.value):
 
     Parameters
     ----------
-    
+
     r : float
         Distance from the centre
     theta : float
@@ -123,7 +123,7 @@ def christoffels(r, theta, M, c=constant.c.value, G=constant.G.value):
 
     Parameters
     ----------
-    
+
     r : float
         Distance from the centre
     theta : float

